File: src/Kisi.c
/**
*
* @author Elvin Valiyev - elvin.valiyev@sakarya.edu.tr-B231210565
* @since 18.05.2025
* <p>
*  Kisi yapisi
* </p>
*
*/

#include "Kisi.h"


typedef struct sKisi{
int yas;
char isim[100];
int kalanOmur;
char bulunduguUzayAraciAdi[100];
} Kisi;

Kisi* Construct_Kisi(const char* isim,const char*yas,const char*kalanOmur,const char* uzayAraciAdi){
Kisi *kisi=(Kisi*)malloc(sizeof(Kisi));
if(kisi==NULL){
    printf("Heap te yer ayrilamadi!");
    exit(EXIT_FAILURE);
}
strncpy(kisi->isim,isim,sizeof(kisi->isim)-1);
kisi->yas=atoi(yas);
kisi->kalanOmur=atoi(kalanOmur);
strncpy(kisi->bulunduguUzayAraciAdi,uzayAraciAdi,sizeof(kisi->bulunduguUzayAraciAdi)-1);

return kisi;
}

int getKalanOmur(Kisi* kisi){
    return kisi->kalanOmur;
}

void setKalanOmur(Kisi* kisi,int komur){
    if(komur>=0){
        kisi->kalanOmur=komur;
    }
    else{
        printf("%s","Hata,Kisinin kalan omru negatif olarak atanmaya calisildi");
    }

}

const char* getKisiUzayAraciAdi(Kisi* kisi){
    return kisi->bulunduguUzayAraciAdi;
}

const char* getKisiAdi(Kisi* kisi){
    return kisi->isim;
}

int getKisiYas(Kisi* kisi){
    return kisi->yas;
}

void kisiOmurAzalt(Kisi *kisi){
    setKalanOmur(kisi,kisi->kalanOmur-1);
} 

void Destruct_Kisi(Kisi* kisi){
    free(kisi);
}

File: src/odev1/Kisi.java
package odev1;
/**
*
* @author Elvin Valiyev - elvin.valiyev@sakarya.edu.tr
* @since 31.03.2025
* <p>
* Kisilere ait veriler bu sinif yapisina getirilir
* </p>
*
*/



public class Kisi 
{
	
	
	
	private int yas;
	private String isim;
	private int kalanOmur;
	private String bulunduguUzayAraciAdi;
	
	public int getKalanOmur() 
	{
		return kalanOmur;
	}
	public void setKalanOmur(int ko) 
	{
		if(ko>=0) this.kalanOmur=ko;
	}
	public String getUzayAraciAdi() 
	{
		return this.bulunduguUzayAraciAdi;
	}
	public String getKisiAdi()
	{
		return this.isim;
	}
	Kisi(String a, String p, String r,String m)
	{
		this.isim=a;
		this.yas=Integer.parseInt(p);
		this.kalanOmur=Integer.parseInt(r);
		this.bulunduguUzayAraciAdi=m;
		
	}
	public int getYas() 
	{
		return this.yas;
	}
	public String getYasString() 
	{
		return Integer.toString(this.yas);
	}

	
	public static void kisiOmurAzalt(Kisi k) 
	{
	    k.setKalanOmur(k.getKalanOmur() - Simulasyon.mode);
	    if(k.getKalanOmur()/Simulasyon.mode==0) 
	    {
	    	k.setKalanOmur(0);
	    }
	}

}
